src/extensionSlot.ts

```diff
@@ -29,16 +32,21 @@ export function createExtensionSlot<T>(key: SlotKey<T>, host: AppHost, declaring
         getItems,
         getSingleItem,
         getItemByName,
-        discardBy
+        discardBy,
+        subscribe
     }
 
     function contribute(fromShell: Shell, item: T, condition?: ContributionPredicate): void {
+        const originalContributionCount = items.length
         items.push({
             shell: fromShell,
             contribution: item,
             condition: condition || alwaysTrue,
             uniqueId: _.uniqueId(`${fromShell.name}_extItem_`)
         })
+        if (items.length !== originalContributionCount) {
```

Comment:
I believe you don't really need that 🙂

Reply:
interesting :) I guess you are right. Still needed in the `discardBy` though, right (as it might not result in a real change to the items)

Reply:
Done